src/main/kotlin/org/jetbrains/research/tasktracker/config/ide/MainIdeConfig.kt

```diff
@@ -26,6 +28,14 @@ class MainIdeConfig : BaseConfig {
                         logger.info("Building config for inspections...")
                         mainIdeConfig.inspectionConfig = InspectionConfig.buildConfig(configFile)
                     }
+
+                    configFile.name.startsWith(SettingsConfig.CONFIG_FILE_PREFIX) -> {
+                        require(mainIdeConfig.settingsConfig == null) {
+                            "The inspections config was already parsed"
+                        }
+                        logger.info("Building config for completion...")
+                        mainIdeConfig.settingsConfig = SettingsConfig.buildConfig(configFile)
+                    }
```

Comment:
Let's create a new helper function to repeat code duplication:
```kotlin
        private fun <T : BaseConfig> buildConfig(
            config: T?,
            configName: String,
            configFile: File,
            configBuilder: (File) -> T
        ): T {
            require(config == null) {
                "The $configName config was already parsed"
            }
            logger.info("Building $configName config...")
            return configBuilder(configFile)
        }

....
                    configFile.name.startsWith(InspectionConfig.CONFIG_FILE_PREFIX) -> {
                        mainIdeConfig.inspectionConfig = buildConfig(
                            mainIdeConfig.inspectionConfig,
                            "inspections",
                            configFile,
                            InspectionConfig::buildConfig
                        )
                    }
```

Reply:
Also it is even better to move configName into the BaseConfig class with the default value "" and override it in the necessary configs